pyaerocom/colocation/colocated_data.py

```diff
@@ -128,13 +128,11 @@ def validate_data(self):
             # make sure path is str instance
             self.data = str(self.data)
         if isinstance(self.data, str):
-            assert self.data.endswith("nc"), ValueError(
-                "Invalid data filepath str, must point to a .nc file"
-            )
+            if not self.data.endswith("nc"):
+                raise ValueError("Invalid data filepath str, must point to a .nc file")
             self.open(self.data)
         elif isinstance(self.data, xr.DataArray):
```

Comment:
I think these can be flipped to reduce nesting: if isinstance(ndarray) -> make_DataArray; then ensure_correct_dimensions.

This would also simplify ensure_correct_dimensions to not have to handle ndarray